swarm/src/connection.rs

```diff
@@ -354,8 +354,11 @@ where
                         if let Some(new_duration) = deadline.checked_duration_since(Instant::now())
                         {
                             let effective_keep_alive = max(new_duration, *idle_timeout);
-
-                            timer.reset(effective_keep_alive)
+                            
+                            let now = Instant::now();
+                            let safe_effective_keep_alive = checked_add_fraction(now, effective_keep_alive);
+                            
+                            timer.reset(safe_effective_keep_alive)
```

Comment:
How about the following, computing `now` once and adding a comment.

``` rust
                    if *deadline != t {
                        *deadline = t;
                        let now = Instant::now();
                        if let Some(new_duration) = deadline.checked_duration_since(now) {
                            let effective_keep_alive = max(new_duration, *idle_timeout);

                            // `Delay::reset` panics if `now + effective_keep_alive` is `> u64::MAX`.
                            let safe_effective_keep_alive = checked_add_fraction(now, effective_keep_alive);
                            timer.reset(safe_effective_keep_alive)
```

Reply:
Looks good to me. And I will update the `CHANGELOG.md` then.